.github/workflows/examples-providers.yml

```diff
@@ -26,42 +26,31 @@ env:
 
 jobs:
   build-wash-cli:
-    name: build wash-cli
+    name: build wash-cli (${{ matrix.wash-version }})
     runs-on: ubuntu-22.04
     strategy:
       matrix:
         wash-version:
           - 0.37.0
           - current
     steps:
-      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
-        if: ${{ matrix.wash-version == 'current' }}
-      - run: rustup show
-      - uses: Swatinem/rust-cache@82a92a6e8fbeee089604da2575dc567ae9ddeaab
-        with:
-          shared-key: "ubuntu-22.04-shared-cache"
-      - name: install wash (previous version)
-        if: ${{ matrix.wash-version != 'current' }}
-        uses: taiki-e/install-action@2c3c8dfabf5933531aca0613c6b0c173fa51745e
-        with:
-          tool: wash-cli@${{ matrix.wash-version }}
-      - name: build wash (current)
-        if: ${{ matrix.wash-version == 'current' }}
-        run: cargo build -p wash-cli --release
-      - uses: actions/upload-artifact@6f51ac03b9356f520e9adb1b1b7802705f340c2b
-        if: ${{ matrix.wash-version == 'current' }}
-        with:
-          name: wash-${{ matrix.wash-version }}
-          path: ./target/release/wash
-      - uses: actions/upload-artifact@6f51ac03b9356f520e9adb1b1b7802705f340c2b
-        if: ${{ matrix.wash-version != 'current' }}
-        with:
-          name: wash-${{ matrix.wash-version }}
-          path: ~/.cargo/bin/wash
+    - name: install wash (${{ matrix.wash-version }})
+      if: ${{ matrix.wash-version && matrix.wash-version != 'current' }}
+      uses: taiki-e/install-action@acd25891978b4cdaebd139d3efef606d26513b14 # v2.47.0
+      with:
+        tool: wash@${{ matrix.wash-version }}
+
+    - name: install wash (current)
+      if: ${{ matrix.wash-version && matrix.wash-version == 'current' }}
+      uses: taiki-e/cache-cargo-install-action@caa6f48d18d42462f9c30df89e2b4f71a42b7c2c # v2.0.1
+      with:
+        tool: wash-cli
+        git: https://github.com/${{ github.repository }}
```

Comment:
You could use [`${{ github.repositoryUrl }}`](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#example-contents-of-the-github-context) for this